inginious/frontend/plugins/analytics/api/stacked_bar_plot_api.py

```diff
@@ -0,0 +1,117 @@
+import web
+
+from inginious.frontend.pages.api._api_page import APIError
+from inginious.frontend.plugins.utils.superadmin_utils import SuperadminAPI
+from ..utils import get_api_query_parameters
+
+
+def get_course_ids(data):
+    """" Return a list of course ids """
+    return list(map(lambda course: course["course"], data))
+
+
+def create_y_data_list(num_courses, num_services):
+    """ create zero list to represent y data """
+    return [[0] * num_services for i in range(num_courses)]
+
+
+def get_visit_values(data, num_course, service_names):
+    """ return the y axis data  """
+    y_data_list = create_y_data_list(num_course, len(service_names))
+    for i in range(len(data)):
+        course = data[i]
+        for service in course["data"]:
+            index = service_names.index(service["service"])
+            y_data_list[i][index] = service["visits"]
+    return y_data_list
+
+
+class StackedBarPlotAPI(SuperadminAPI):
+    """ API to get a data to plot the stacked bar """
+    def API_GET(self):
+        """ Get request """
+        self.check_superadmin_rights()
+        input_dict = web.input()
+
+        try:
+            query_parameters = get_api_query_parameters(input_dict)
+        except APIError as error:
+            return error.status_code, {"error": error.return_value}
+
+        data = list(self.get_visits_data(query_parameters))
+
+        course_names = get_course_ids(data)
```

Comment:
why the variable is named `course_names` and the function returns the ids? Try to have consistency with the names of variables and functions.

Reply:
yes, in the last moment I decided to change the name of the function `get_course_names` to `get_course_ids` and I forgot to change the names of the variables